### PR TITLE
Fix workarea display bug

### LINKED ISSUE #4881: Workareas display bug with some add-on buildings
1. Start a new SP game on the map *The Pass Through The Mountains* with both players set to *Europeans*, the *Endless Game Fogless* win condition, and *Peaceful Mode* enabled.
2. Ensure that the option to indicate overlapping workareas is turned on.
3. Click on the building spot located 4 nodes W and 2 SW of the HQ's flag.
4. Select the small buildings tab. Point on the first building (basic well) and slowly move the mouse to the right over all buildings in the row (up to basic quarry).
5. Right-click to close the fieldaction window.
6. Some workareas (or parts of them) remain permanently visible. Some of them should never have been displayed in the first place.
shot0001

Version 1.0~git25311[d15ecd3@master] (Release) with Europeans 1.1.0

### PR DESCRIPTION
**Pull request by matthiakl**
Fixes #4881 
- No duplicate workareas
- Hide unowned workareas

### CODE REVIEW

File: src/wui/fieldaction.cc [hunk 1]
```diff
@@ -990,19 +991,27 @@ void FieldActionWindow::building_icon_mouse_in(const Widelands::DescriptionIndex
...
+							if (mr.location() == t.node || mr.location() == map.br_n(t.node) ||
+							    mr.location() == coords || main_region.count(t) != 0u) {
+								if (descr.type() == Widelands::MapObjectType::PRODUCTIONSITE) {
+									colors[t] = positive ? kOverlapColorPositive : kOverlapColorNegative;
+								} else {
+									colors[t] = kOverlapColorDefault;
+								}
+							} else {
+								colors[t] = kOverlapColorPale;
+							}
```
Comment (matthiakl) [inline src/wui/fieldaction.cc:1008]: Only refactored for better readability. No logic changes.

**Review by Noordfrees — CHANGES_REQUESTED** [review]
This removes too many workareas. When placing a building with a conquer radius we also want to highlight all intersecting workareas of enemy buildings with conquer radii, and this change would disable this feature.

**Comment by matthiakl** [comment]
Ah, ok. I didn't know it for military site. On a second look the ownership check was present for production sites, but missing for construction sites.

**Review by Noordfrees — COMMENTED** [review]
Now when I place the constructionsite for a lumberjack's hut the adjacent constructionsite for a ranger's hut is not highlighted

**Comment by matthiakl** [comment]
> Now when I place the constructionsite for a lumberjack's hut the adjacent constructionsite for a ranger's hut is not highlighted

Fixed. clang-tidy reported an equal condition, but there was one small important difference.

**Review by Noordfrees — APPROVED** [review]

**Comment by matthiakl** [comment]
Thanks for the review :)

**Pull request merged**